Sources/Apollo/RecordSet.swift

```diff
@@ -10,8 +10,20 @@ public struct RecordSet {
     storage[record.key] = record
   }
 
-  public mutating func clear() {
-    storage.removeAll()
+  public mutating func clear(_ clearingPolicy: CacheClearingPolicy = .allRecords) {
+    switch clearingPolicy {
+    case let .first(count):
+      self.storage = Dictionary(self.storage.dropFirst(count))
```

Comment:
As far as I know, `Dictionary` has no guaranteed order, so I don't think we can use `dropFirst`/`dropLast` here.